### PR TITLE
feat(ecs): Ensure ECS containers have read-only access to root filesystems

### PR DESCRIPTION
**Pull request by MarioRgzLpz**
### Context

This check verifies whether Amazon ECS containers are restricted to read-only access for mounted root filesystems. The check fails if the readonlyRootFilesystem parameter is set to false or is missing in the container definition within the task definition. This check evaluates only the latest active revision of an Amazon ECS task definition.

Enabling this option reduces security attack vectors since the container instance's filesystem cannot be tampered with or written to unless it has explicit read-write permissions on its filesystem folder and directories. This control also adheres to the principle of least privilege.

Fixed `ecs_task_definitions_host_networking_mode_users` name and mocked service and client.

### Description

Change `ecs_task_definitions_user_and_container_for_host_mode` to `ecs_task_definitions_host_networking_mode_users` and fix mocks.  Added new check `ecs_task_definitions_container_readonly` with respective unit tests. Change Container model in `ecs_service` to add atributte` readonly_rootfilesystems.`

### Checklist

- Are there new checks included in this PR? Yes 
    - If so, do we need to update permissions for the provider?
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings
- [x] Review if backport is needed.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly.metadata.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
+{
+  "Provider": "aws",
+  "CheckID": "ecs_task_definitions_containers_readonly",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly.metadata.json]:
```suggestion
  "CheckID": "ecs_task_definitions_containers_readonly_access",
```

File: prowler/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def execute(self):
+        findings = []
+        for task_definition in ecs_client.task_definitions.values():
+            report = Check_Report_AWS(self.metadata())
+            report.region = task_definition.region
+            report.resource_id = f"{task_definition.name}:{task_definition.revision}"
+            report.resource_arn = task_definition.arn
+            report.resource_tags = task_definition.tags
+            report.status = "PASS"
+            report.status_extended = f"ECS task definition {task_definition.name} containers have read-only root filesystems."
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly.py]:
```suggestion
            report.status_extended = f"ECS task definition {task_definition.name} does not have containers with write access to the root filesystems."
```

File: tests/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly_test.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+from prowler.providers.aws.services.ecs.ecs_service import (
+    ContainerDefinition,
+    TaskDefinition,
+)
+from tests.providers.aws.utils import AWS_ACCOUNT_NUMBER, AWS_REGION_US_EAST_1
+
+task_name = "test-task-readonly"
+task_revision = "1"
+container_name = "test-container"
+task_arn = f"arn:aws:ecs:{AWS_REGION_US_EAST_1}:{AWS_ACCOUNT_NUMBER}:task-definition/{task_name}:{task_revision}"
```
Comment (MrCloudSec) [inline tests/providers/aws/services/ecs/ecs_task_definitions_containers_readonly/ecs_task_definitions_containers_readonly_test.py]: Please, use uppercase letters for global variables.

File: tests/providers/aws/services/ecs/ecs_task_definitions_host_networking_mode_users/ecs_task_definitions_host_networking_mode_users_test.py [hunk 4]
```diff
@@ -5,13 +5,9 @@
...
-    AWS_REGION_US_EAST_1,
-    set_mocked_aws_provider,
-)
+from tests.providers.aws.utils import AWS_ACCOUNT_NUMBER, AWS_REGION_US_EAST_1
 
-task_name = "test-task"
+task_name = "test-task-hostmode"
 task_revision = "1"
 container_name = "test-container"
 task_arn = f"arn:aws:ecs:{AWS_REGION_US_EAST_1}:{AWS_ACCOUNT_NUMBER}:task-definition/{task_name}:{task_revision}"
```
Comment (MrCloudSec) [inline tests/providers/aws/services/ecs/ecs_task_definitions_host_networking_mode_users/ecs_task_definitions_host_networking_mode_users_test.py]: Please, use uppercase letters for global variables.

**Comment by MarioRgzLpz** [comment]
@sergargar requested changes added -> [6090658](https://github.com/prowler-cloud/prowler/pull/5168/commits/60906584eac7707d779928f24b6db694c0710929)

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**